File: Sort-Algorithms-2.0/Python/Bubble_Sort.py
import os
from time import time

import random

def BubbleSort(V):
    for i in range(0,len(V),1):
        for j in range(0,len(V)-1,1):
            if V[j]>V[j+1]:
                temp = V[j]
                V[j] = V[j+1]
                V[j+1] = temp

def crear_arreglo(A,n):
    for i in range(n):
        A.append(random.randint(1,100000))
        

if __name__=='__main__':
    A=[100000]
    f=open("tiempos_bubble_p.txt","w")
    for i in range(10000,100001,10000):
        crear_arreglo(A,i)
        tiempo_inicial=time()
        BubbleSort(A)
        tiempo_final=time()
        tiempo_ejecucion=tiempo_final-tiempo_inicial
        f.write(str(tiempo_ejecucion))
        print("tiempo de ejecucion: ",tiempo_ejecucion)
    f.close()
        

File: Sort-Algorithms-2.0/C++/Bubble_Sort.cpp
#include<iostream>
#include<time.h>
#include<fstream>
#include<stdlib.h>
using namespace std;

void BubbleSort(int *v, int n){
    for(int i=0;i<n;i++){
        for(int j=0;j<n-1;j++){
            if(v[j]>v[j+1]){
                int aux=v[j];
                v[j]=v[j+1];
                v[j+1]=aux;
            }
        }
    }
}

void crear_arreglo(int *v, int n){
    int c;
    srand(time(NULL));

    for(c=0;c<=n;c++){
        v[c]=rand()%100000;

    }
}

int main()
{
    unsigned t0,t1;
    double time;
    int *v=new int[100000];
    ofstream archivo;
    archivo.open("tiempos_bubble_c.txt",ios::out);
    for(int i=10000;i<=100000;i+=10000){
        crear_arreglo(v,i);
        t0=clock();
        BubbleSort(v,i-1);
        t1=clock();
        time=(double(t1-t0)/CLOCKS_PER_SEC);
        cout<<"Tiempo de ejecucion: "<<time<<endl;
        archivo<<time<<endl;
    }

    archivo.close();
    delete[] v;
    return 0;
}
